stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -169,6 +169,7 @@ internal class PaymentSheetViewModel internal constructor(
     fun checkout() {
         _userMessage.value = null
         _processing.value = true
+        _viewState.value = ViewState.PaymentSheet.StartProcessing
```

Comment:
I really like this addition to start the processing state here.  I am not sure of all the paths through the states if the PaymentSelection is Google.   Do we reset the view state in all of those scenarios?

Reply:
The result is a callback to `onGooglePayResult`. On error it calls `resetViewState` and on success it continues through `confirmPaymentSelection`, which is the same flow as using a card. I did not try to reproduce all error cases, we still need a task to show errors and make sure all of them are handled.